helm/vitess/templates/_pmm.tpl

```diff
@@ -0,0 +1,197 @@
+              value: {{ $pmm.server.env.metricsRetention | quote }}
+
+            - name: QUERIES_RETENTION
+              value: {{ $pmm.server.env.queriesRetention | quote }}
+
+            - name: METRICS_MEMORY
+              value: {{ $pmm.server.env.metricsMemory | quote }}
+        
+          command: ["bash"]
+          args:
+            - "-c"
+            - |
+              set -ex
+
+
+              if [ ! -f /pmmdata/vitess-init ]; then
+                  # the PV hasn't been initialized, so copy over default
+                  # pmm-server directories before symlinking
+                  mkdir -p /pmmdata
+
+                  mv /opt/prometheus/data /pmmdata/data
+                  mv /opt/consul-data /pmmdata
+                  mv /var/lib/mysql /pmmdata
+                  mv /var/lib/grafana /pmmdata
+
+                  # initialize the PV and then mark it complete
+                  touch /pmmdata/vitess-init
+              else
+                  # remove the default directories so we can symlink the
+                  # existing PV directories
+                  rm -Rf /opt/prometheus/data
+                  rm -Rf /opt/consul-data
+                  rm -Rf /var/lib/mysql
+                  rm -Rf /var/lib/grafana
+              fi
+
+              # symlink pmm-server paths to point to our PV
+              ln -s /pmmdata/data /opt/prometheus/
+              ln -s /pmmdata/consul-data /opt/
+              ln -s /pmmdata/mysql /var/lib/
+              ln -s /pmmdata/grafana /var/lib/
+              
+              /opt/entrypoint.sh
+
+  volumeClaimTemplates:
+    - metadata:
+        name: pmmdata
+        annotations:
+{{ toYaml $pmm.server.dataVolumeClaimAnnotations | indent 10 }}
+      spec:
+{{ toYaml $pmm.server.dataVolumeClaimSpec | indent 8 }}
+
+{{- end -}}
+
+###################################
+# sidecar container running pmm-client
+###################################
+{{- define "cont-pmm-client" -}}
+{{- $pmm := index . 0 -}}
+{{- $namespace := index . 1 -}}
+
+- name: "pmm-client"
+  image: "vitess/pmm-client:{{ $pmm.pmmTag }}"
+  imagePullPolicy: IfNotPresent
+  volumeMounts:
+    - name: vtdataroot
+      mountPath: "/vtdataroot"
+  ports:
+    - containerPort: 42001
+      name: query-data
+    - containerPort: 42002
+      name: mysql-metrics
+
+  securityContext:
+    # PMM requires root privileges
+    runAsUser: 0
+
+  resources:
+{{ toYaml $pmm.client.resources | indent 4 }}
+
+  command: ["bash"]
+  args:
+    - "-c"
+    - |
+      set -ex
+
+      mkdir -p /vtdataroot/pmm
+
+      # redirect logs to PV
+      ln -s /vtdataroot/pmm/pmm-mysql-queries-42001.log /var/log/pmm-mysql-queries-0.log
+      ln -s /vtdataroot/pmm/pmm-mysql-metrics-42002.log /var/log/pmm-mysql-metrics-42002.log
+
+
+      # --force is used because the pod ip address may have changed
+      pmm-admin config --server pmm.{{ $namespace }} --force
+
+      # each of these creates a systemd service
```

Comment:
It seems weird to install a systemd service in a container. Can we not just run the server directly?

Reply:
Their whole setup is very opinionated and there isn't a way around this. You have to use their `pmm-admin` client and it doesn't have an option to run another way.